addons/mrp_landed_costs/tests/test_stock_landed_costs_mrp.py

```diff
@@ -179,3 +179,51 @@ def test_landed_cost_on_mrp_02(self):
         # Check that he can validate the landed cost without an access error
         landed_cost.with_user(stock_manager).button_validate()
         self.assertEqual(landed_cost.state, 'done')
+
+    def test_unbuild_MO_with_landed_cost(self):
+        consu_comp = self.env['product.product'].create({
+            'name': 'consu comp',
+            'type': 'consu',
+            'standard_price': 3.0,
+        })
+        final_prod = self.env['product.product'].create({
+            'name': 'final prod',
+            'type': 'product',
+            'categ_id': self.categ_all.id
+        })
+        final_prod.categ_id.property_cost_method = 'average'
+        bom = self.env['mrp.bom'].create({
+            'product_id': final_prod.id,
+            'product_tmpl_id': final_prod.product_tmpl_id.id,
+            'product_uom_id': self.uom_unit.id,
+            'product_qty': 1.0,
+            'type': 'normal',
+        })
+        self.env['mrp.bom.line'].create({
+            'bom_id': bom.id,
+            'product_id': consu_comp.id,
+            'product_qty': 10,
+        })
+
+        mo = self.env['mrp.production'].create({
+            'product_qty': 1.0,
+            'bom_id': bom.id,
+        })
+        mo.action_confirm()
+        mo.button_mark_done()
+
+        lc_form = Form(self.env['stock.landed.cost'])
+        lc_form.target_model = 'manufacturing'
+        lc_form.mrp_production_ids = mo
+        with lc_form.cost_lines.new() as cost_line:
+            cost_line.product_id = self.landed_cost
+            cost_line.price_unit = 8
+        lc = lc_form.save()
+        lc.compute_landed_cost()
+        lc.button_validate()
+
+        action = mo.button_unbuild()
+        wizard = Form(self.env[action['res_model']].with_context(action['context']))
+        wizard.product_qty = 1
+        unbuild = wizard.save()
+        unbuild.action_validate()
```

Comment:
No assert at the end, it makes the test quite weird
I know you want to ensure that the case is simply working, but you can still assert few things, so that you can ensure that on top of a working case, it also gives correct values